src/AutoRest.CSharp/Mgmt/Decorator/RemoveOperationGroups.cs

```diff
@@ -0,0 +1,199 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using AutoRest.CSharp.Input;
+using AutoRest.CSharp.Mgmt.AutoRest;
+using AutoRest.CSharp.Output.Models.Types;
+
+namespace AutoRest.CSharp.Mgmt.Decorator
+{
+    internal static class RemoveOperationGroups
+    {
+        private static HashSet<Schema> _schemasToOmit = new HashSet<Schema>();
+        private static HashSet<Schema> _schemasStillUsed = new HashSet<Schema>();
+
+        public static void OmitOperationGroups(CodeModel codeModel, BuildContext<MgmtOutputLibrary> context)
+        {
+            var operationGroupsToOmit = context.Configuration.MgmtConfiguration.OperationGroupsToOmit;
+            if (operationGroupsToOmit != null)
+            {
+                var omitSet = operationGroupsToOmit.ToHashSet();
+                for (int i = 0; i < codeModel.OperationGroups.Count; i++)
```

Comment:
omittedOG = codeModel.OperationGroups.Select(og => omitSet.Contains(og.Key));

codeModel.OperationGroups. = codeModel.OperationGroups.Select(og => !omitSet.Contains(og.Key));

codeModel.OperationGroups.Foreach(og => AddNonOmmited(og));

foreach (var og in ommittedOG)
{
   Line 32-35
}